core/ecschema-locaters/src/SchemaFileLocater.ts

```diff
@@ -193,7 +316,49 @@ export abstract class SchemaFileLocater {
    * @param matchType The SchemaMatchType.
    * @param format The type of file that the schema key refers to. json or xml
    */
-  protected findEligibleSchemaKeys(desiredKey: SchemaKey, matchType: SchemaMatchType, format: string): FileSchemaKey[] {
+  protected async findEligibleSchemaKeys(desiredKey: SchemaKey, matchType: SchemaMatchType, format: string): Promise<FileSchemaKey[]> {
+    const foundFiles = new Array<FileSchemaKey>();
+
+    let twoVersionSuffix: string;
+    let threeVersionSuffix: string;
+    const readVersion = desiredKey.readVersion.toString();
+    const writeVersion = desiredKey.writeVersion.toString();
+    const minorVersion = desiredKey.minorVersion.toString();
+
+    if (matchType === SchemaMatchType.Latest) {
+      twoVersionSuffix = (`.*.*.ecschema.${format}`);
+      threeVersionSuffix = (`.*.*.*.ecschema.${format}`);
+    } else if (matchType === SchemaMatchType.LatestWriteCompatible) {
+      twoVersionSuffix = formatString(`.{0}.*.ecschema.${format}`, padStartEx(readVersion, 2, "0"));
+      threeVersionSuffix = formatString(`.{0}.{1}.*.ecschema.${format}`, padStartEx(readVersion, 2, "0"), padStartEx(writeVersion, 2, "0"));
+    } else if (matchType === SchemaMatchType.LatestReadCompatible) {
+      twoVersionSuffix = formatString(`.{0}.*.ecschema.${format}`, padStartEx(readVersion, 2, "0"));
+      threeVersionSuffix = formatString(`.{0}.*.*.ecschema.${format}`, padStartEx(readVersion, 2, "0"));
+    } else {
+      twoVersionSuffix = formatString(`.{0}.{1}.ecschema.${format}`, padStartEx(readVersion, 2, "0"), padStartEx(writeVersion, 2, "0"));
+      threeVersionSuffix = formatString(`.{0}.{1}.{2}.ecschema.${format}`, padStartEx(readVersion, 2, "0"), padStartEx(writeVersion, 2, "0"), padStartEx(minorVersion, 2, "0"));
+    }
+
+    const twoVersionExpression = desiredKey.name + twoVersionSuffix;
+    const threeVersionExpression = desiredKey.name + threeVersionSuffix;
```

Comment:
Same, here pull this code out into it's own method to avoid duplication

Reply:
I created two methods named `getTwoVersionExpression` and `getThreeVersionExpression`. However, I think it is possible to merge both methods into a single one that would return an array of VersionExpressions. What's your opinion on this?